config/config_helpers.go

```diff
@@ -524,13 +534,25 @@ func sopsDecryptFile(params []string, include *IncludeConfig, terragruntOptions
 		return "", errors.WithStackTrace(InvalidSopsFormat{SourceFilePath: sourceFile})
 	}
 
+	canonicalSourceFile, err := util.CanonicalPath(sourceFile, terragruntOptions.WorkingDir)
+	if err != nil {
+		return "", errors.WithStackTrace(err)
+	}
+
+	if val, ok := sopsCache[canonicalSourceFile]; ok {
+		return val, nil
+	}
+
 	rawData, err := decrypt.File(sourceFile, format)
 	if err != nil {
 		return "", errors.WithStackTrace(err)
 	}
 
 	if utf8.Valid(rawData) {
-		return string(rawData), nil
+		var value string
+		value := string(rawData)
```

Comment:
This won't work. You need to remove the declaration, as `:=` sets a value and declares it all in one line.

```suggestion
		value := string(rawData)
```